app/controllers/shared_links_controller.rb

```diff
@@ -0,0 +1,39 @@
+class SharedLinksController < ApplicationController
+  before_action :redirect_to_root_unless_signed_in, :shared_link_params, :set_term
+
+  def create
+    return redirect_to root_path if current_user.has_available_link_on?(@term)
+
+    token = SecureRandom.hex(32)
+    shared_link = SharedLink.new(user: current_user, expired_at: 30.minute.after, on: @term, token: token)
```

Comment:
```suggestion
    shared_link = current_user.shared_links.build(on: @term)
```

- `current_user` 起点にしておくと認可の処理をミスしにくくなる
- `expired_at` もモデル側でよしなにするとよさそう